middleware/proxy/reverseproxy.go

```diff
@@ -153,8 +155,37 @@ func (p *ReverseProxy) ServeHTTP(rw http.ResponseWriter, req *http.Request, extr
 
 	copyHeader(rw.Header(), res.Header)
 
-	rw.WriteHeader(res.StatusCode)
-	p.copyResponse(rw, res.Body)
+	if res.StatusCode == HTTPSwitchingProtocols && outreq.Header.Get("Upgrade") == "websocket" {
```

Comment:
Actually I was doing some testing/debugging on this with Swift client library I assist with and it looks like using caddy as a reverse proxy, I'm still getting the headers and status 101 back to the client That makes sense, as our client library would have failed to connect otherwise.

```
{
    Connection = Upgrade;
    "Sec-WebSocket-Accept" = "aOf2eAMNAN875GUJeXOs6dn05XE=";
    Upgrade = websocket;
}
```

Looking at the code again, that should be the case, since we hijack the connection before writing back to the client and then just write the response in the raw tcp stream back to the client.

```go
outreq.Write(backendConn)
```

We write the original request the client sent to the server and then use 

```go
io.Copy(conn, backendConn)
```

to write back whatever the response is to the client. The code does seem a little unclear on that right now. Maybe it would make sense to move 

```go
	for _, h := range hopHeaders {
		res.Header.Del(h)
	}

	copyHeader(rw.Header(), res.Header)
```

to the else statement, since we never use them in a web socket connection anyway?